File: java/claude/P8.java
// Task p8: atoi, argc, argv explanation
public class P8 {
    public static void main(String[] args) {
        // argc is the number of arguments passed to the program
        // argv is an array of strings; each string is one of the arguments
        
        int argc = args.length;
        String[] argv = args;
        
        System.out.println("argc (number of arguments): " + argc);
        System.out.println("\nargv (array of arguments):");
        
        for (int i = 0; i < argv.length; i++) {
            System.out.println("argv[" + i + "]: " + argv[i]);
        }
        
        // Example: if the user ran: java P8 foo bar baz
        // argc would be 3
        // argv[0] would be 'foo'
        // argv[1] would be 'bar'
        // argv[2] would be 'baz'
    }
}

File: javascript/claude/p8.js
// Task p8: atoi, argc, argv explanation
// argc is the number of arguments passed to the program
// argv is an array of strings; each string is one of the arguments

const argc = process.argv.length;
const argv = process.argv;

console.log(`argc (number of arguments): ${argc}`);
console.log("\nargv (array of arguments):");

argv.forEach((arg, i) => {
  console.log(`argv[${i}]: ${arg}`);
});

// Example: if the user ran: node p8.js foo bar baz
// argc would be 5 (node, script, foo, bar, baz)
// argv[0] would be 'node'
// argv[1] would be 'p8.js' (the program)
// argv[2] would be 'foo'
// argv[3] would be 'bar'
// argv[4] would be 'baz'
